telegram/_message.py

```diff
@@ -1539,10 +1547,113 @@ def _quote(self, quote: Optional[bool], reply_to_message_id: Optional[int]) -> O
             else:
                 default_quote = None
             if (default_quote is None and self.chat.type != Chat.PRIVATE) or default_quote:
-                return self.message_id
+                return ReplyParameters(self.message_id)
 
         return None
 
+    def compute_quote_position_and_entities(
+        self, quote: str, index: Optional[int] = None
+    ) -> Tuple[int, Tuple[MessageEntity, ...]]:
```

Comment:
needs a docstring. This should also point out that `quote` is expected to plain text without formatting entities as explained in https://github.com/python-telegram-bot/python-telegram-bot/issues/4033#issuecomment-1874358806 as well as list the `{Runtime, Value}Error` with reasons